daily_workflow.py

```
@@ -1080,6 +1054,38 @@ def _scan_body(cfg: UltimateConfig) -> tuple:
+                import pandas_market_calendars as mcal
+
+                nse_calendar = mcal.get_calendar("NSE")
+                valid_days = nse_calendar.valid_days(
+                    start_date=expected_session_date - pd.Timedelta(days=366),
+                    end_date=expected_session_date,
+                )
+                if len(valid_days) > 0:
+                    expected_session_date = pd.Timestamp(valid_days[-1]).tz_convert("Asia/Kolkata").tz_localize(None).normalize()
+                else:
+                    expected_session_date = (expected_session_date - pd.offsets.BDay(1)).tz_localize(None)
+            except Exception:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)